packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -222,10 +215,174 @@ describe('<DataGrid /> - Filter', () => {
+      },
+      {
+        operator: 'onOrBefore',
+        value: '2001-01-01',
+        expected: [
+          '12/1/2000, 12:00:00 AM',
+          '12/1/2000, 8:30:00 AM',
+          '1/1/2001, 12:00:00 AM',
+          '1/1/2001, 8:30:00 AM',
+        ],
+      },
+    ].forEach(({ operator, value, expected }) => {
+      it(`should work with dates at different hours, operator ${operator}`, () => {
+        render(
+          <TestCase
+            value={value}
+            operator={operator}
+            rows={[
+              {
+                id: 3,
+                brand: new Date(2000, 11, 1), // 12/1/2000, 12:00:00 AM
+              },
+              {
+                id: 4,
+                brand: new Date(2000, 11, 1, 8, 30), // 12/1/2000, 8:30:00 AM
+              },
+              {
+                id: 5,
+                brand: new Date(2001, 0, 1), // -> 1/1/2001, 12:00:00 AM
+              },
+              {
+                id: 6,
+                brand: new Date(2001, 0, 1, 8, 30), // 1/1/2001, 08:30:00 AM
+              },
+              {
+                id: 7,
+                brand: new Date(2002, 0, 1), // -> 1/1/2002, 12:00:00 AM
+              },
+              {
+                id: 8,
+                brand: new Date(2002, 0, 1, 8, 30), // 1/1/2002, 08:30:00 AM
+              },
+            ]}
+            columns={[
+              {
+                field: 'brand',
+                type: 'date',
+                valueFormatter: (params) => params.value.toLocaleString('en-US'),
+              },
+            ]}
+          />,
+        );
+        expect(getColumnValues()).to.deep.equal(expected);
+      });
+    });
+  });
+
+  describe('dateTime operators', () => {
+    [
+      {
+        operator: 'is',
+        value: '2000-12-01T08:30',
+        expected: ['12/1/2000, 8:30:15 AM'],
+      },
+      {
+        operator: 'not',
+        value: '2000-12-01T08:30',
+        expected: ['1/1/2001, 8:30:15 AM', '1/1/2002, 8:30:15 AM'],
+      },
+      {
+        operator: 'after',
+        value: '2001-01-01T08:30',
+        expected: ['1/1/2002, 8:30:15 AM'],
+      },
+      {
+        operator: 'onOrAfter',
+        value: '2001-01-01T08:30',
+        expected: ['1/1/2001, 8:30:15 AM', '1/1/2002, 8:30:15 AM'],
+      },
+      {
+        operator: 'before',
+        value: '2001-01-01T08:30',
+        expected: ['12/1/2000, 8:30:15 AM'],
+      },
+      {
+        operator: 'onOrBefore',
+        value: '2001-01-01T08:30',
+        expected: ['12/1/2000, 8:30:15 AM', '1/1/2001, 8:30:15 AM'],
+      },
+    ].forEach(({ operator, value, expected }) => {
+      it(`should work correctly with operator=${operator}`, () => {
+        render(
+          <TestCase
+            value={value}
+            operator={operator}
+            rows={[
+              {
+                id: 3,
+                brand: new Date(2000, 11, 1, 8, 30, 15, 20), // 12/1/2000, 8:30:15 AM
```

Comment:
Can we add a test with null, undefined, and empty values as well as dates?

Reply:
I added more tests with undefined and empty values. We can't test for null or dates because the definition of the filter model doesn't allow them.

Reply:
I meant undefined value for a cell in a row, along with some dates, I believe you just put undefined on the filter value.

Reply:
Updated with more test cases with null and empty inside the `rows` prop.